### PR TITLE
Codacy issue resolution for 3 HA files

### PR DESCRIPTION
**Pull request by rahul-hatwar**
Signed-off-by: Rahul Hatwar <rahulchandrakant.hatwar@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by rahul-hatwar** [comment]
{'nodeid': 'tests/s3/versioning/test_versioning_get_head_object.py::TestVersioningGetHeadObject::test_get_head_object_versioned_bucket_32731', 'test_id': 'TEST-32731', 'marks': ['s3_ops']}, {'nodeid': 'tests/s3/versioning/test_versioning_get_head_object.py::TestVersioningGetHeadObject::test_get_head_object_versioned_bucket_32729', 'test_id': 'TEST-32729', 'marks': ['s3_ops']}, {'nodeid': 'tests/s3/versioning/test_versioning_get_head_object.py::TestVersioningGetHeadObject::test_get_head_versioned_object_invalid_32727', 'test_id': 'TEST-32727', 'marks': ['s3_ops']}, {'nodeid': 'tests/s3/versioning/test_versioning_get_head_object.py::TestVersioningGetHeadObject::download_and_check', 'test_id': '', 'marks': []}, {'nodeid': 'tests/s3/versioning/test_versioning_put_object.py::TestVersioningPutObject::test_put_object_preexisting_32724', 'test_id': 'TEST-32724', 'marks': ['s3_ops']}, {'nodeid': 'tests/s3/versioning/test_versioning_put_object.py::TestVersioningPutObject::test_put_object_versioning_enabled_32728', 'test_id': 'TEST-32728', 'marks': ['s3_ops']}, {'nodeid': 'tests/s3/versioning/test_versioning_put_object.py::TestVersioningPutObject::test_put_object_versioning_suspended_32733', 'test_id': 'TEST-32733', 'marks': ['s3_ops']}, {'nodeid': 'tests/security/test_cortx_port_scanner_kubectl_svc.py::test_cortx_port_scanner_kubectl_svc', 'test_id': 'TEST-34217', 'marks': ['security']}, {'nodeid': 'tests/security/test_cortx_port_scanner_netstat.py::test_cortx_port_scanner_netstat', 'test_id': 'TEST-34218', 'marks': ['security']}] created at /root/rahul/cortx-test-1/log/te_meta.json
Successfully unmounted directory

============================================ 2156 tests collected in 8.40s ============================================

File: libs/ha/ha_common_libs_k8s.py [hunk 1]
```diff
@@ -767,7 +772,7 @@ def start_random_mpu(self, event, s3_data, bucket_name, object_name, file_size,
                 LOGGER.debug("Part : %s", str(p_tag))
                 parts_etag.append({"PartNumber": i, "ETag": p_tag["ETag"]})
                 LOGGER.info("Uploaded part %s", i)
-            except CTException as error:
+            except BaseException as error:
```
Comment (ApurwaMohite) [inline libs/ha/ha_common_libs_k8s.py:775]: This should be CTException only.

File: libs/ha/ha_common_libs_k8s.py [hunk 2]
```diff
@@ -1377,7 +1381,7 @@ def simulate_disk_cvg_failure(self, node_obj, source: str, resource_status: str,
                 config_dict["events"][f"{count}"]["specific_info"] = specific_info
         if delay:
             config_dict["delay"] = delay
-        with open(config_json_file, "w", encoding="utf-8") as outfile:
+        with open(config_json_file, "w") as outfile:
```
Comment (ApurwaMohite) [inline libs/ha/ha_common_libs_k8s.py:1384]: Are you getting error here? encoding is needed i guess.

File: libs/ha/ha_common_libs_k8s.py [hunk 3]
```diff
@@ -1795,91 +1791,3 @@ def failover_pod(self, pod_obj, pod_yaml, failover_node):
...
-                LOGGER.debug("Created and deleted %s bucket successfully", i)
-            except CTException as error:
-                LOGGER.error("Error: %s", error)
-                if event.is_set():
-                    exp_fail.append(bucket_name)
-                else:
-                    failed.append(bucket_name)
-
-        result = (exp_fail, failed)
-        output.put(result)
```
Comment (ApurwaMohite) [inline libs/ha/ha_common_libs_k8s.py:1885]: These both methods are needed. Why are these removed?

File: libs/ha/ha_common_libs_k8s.py [hunk 4]
```diff
@@ -402,7 +402,7 @@ def cortx_start_cluster(self, pod_obj):
         LOGGER.info("Start the cluster")
         resp = pod_obj.execute_cmd(common_cmd.CLSTR_START_CMD.format(self.dir_path),
                                    read_lines=True, exc=False)
-        LOGGER.debug("Cluster start response: %s", resp)
+        LOGGER.info("Cluster start response: %s", resp)
```
Comment (ankita-93) [inline libs/ha/ha_common_libs_k8s.py:405]: debug was good fit here

File: libs/ha/ha_common_libs_k8s.py [hunk 5]
```diff
@@ -109,7 +109,7 @@ def polling_host(self,
                     common_cmd.CMD_VM_INFO.format(
                         self.vm_username, self.vm_password, vm_name))
                 if not vm_info[0]:
-                    LOGGER.error("Unable to get VM power status for %s", vm_name)
+                    LOGGER.info("Unable to get VM power status for %s", vm_name)
```
Comment (ankita-93) [inline libs/ha/ha_common_libs_k8s.py:112]: or debug

File: libs/ha/ha_common_libs.py [hunk 6]
```diff
@@ -261,14 +263,14 @@ def get_iface_ip_list(node_list: list, num_nodes: int):
...
             resp_ip = node_list[0].execute_cmd(
                 common_cmd.CMD_HOSTS, read_lines=True)
-            LOGGER.debug("Response for /etc/hosts: {}".format(resp_ip))
+            LOGGER.debug("Response for /etc/hosts: %s", resp_ip)
             for node in range(num_nodes):
                 for line in resp_ip:
                     if "srvnode-{}.data.private".format(node + 1) in line:
-                        ip = line.split()[0]
-                        private_ip_list.append(ip)
+                        i_p = line.split()[0]
```
Comment (sonalk0209) [inline libs/ha/ha_common_libs.py:270]: instead of i_p can we make it priv_ip maybe?

File: libs/ha/ha_common_libs_k8s.py [hunk 7]
```diff
@@ -1143,14 +1148,14 @@ def get_nw_iface_node_down(host_list: list, node_list: list, node_fqdn: str):
                 resp = system_utils.check_ping(host=node_ip)
                 if not resp:
                     return False, node_ip, node_iface, new_worker_obj
-                else:
-                    return True, node_ip, node_iface, new_worker_obj
+                return True, node_ip, node_iface, new_worker_obj
+            return False, "Worker node and Fqdn of data node not same"
```
Comment (sonalk0209) [inline libs/ha/ha_common_libs_k8s.py:1152]: why do we want to return before for loop concluded?

Comment (rahul-hatwar) [reply libs/ha/ha_common_libs_k8s.py:1152]: Yes this should be after the loop.
Corrected now.

**Review by sonalk0209 — APPROVED** [review]

File: libs/ha/ha_common_libs_k8s.py [hunk 8]
```diff
@@ -1256,12 +1262,9 @@ def get_config_value(pod_obj, pod_list=None):
         if not resp[0]:
             LOGGER.error("Error: Failed to copy cluster.conf to local")
             return False, resp
-        try:
-            with open(local_conf, "r", encoding="utf-8") as file_data:
-                data = yaml.safe_load(file_data)
-        except IOError as error:
-            LOGGER.error("Error: Not able to read local config file")
-            return False, error
+        conf_fd = open(local_conf, 'r')
```
Comment (ankita-93) [inline libs/ha/ha_common_libs_k8s.py:1265]: Why this removed. ? we should use **with** to open file, which will implicitly close the file.  And use of try except was also required incase of file is missing or issue with open.

File: libs/ha/ha_common_libs_k8s.py [hunk 9]
```diff
@@ -1155,14 +1160,14 @@ def get_nw_iface_node_down(host_list: list, node_list: list, node_fqdn: str):
                 resp = system_utils.check_ping(host=node_ip)
                 if not resp:
                     return False, node_ip, node_iface, new_worker_obj
-                else:
-                    return True, node_ip, node_iface, new_worker_obj
+                return True, node_ip, node_iface, new_worker_obj
+        return False, "Worker node and Fqdn of data node not same"
```
Comment (ankita-93) [inline libs/ha/ha_common_libs_k8s.py:1164]: Or can we say, **Could not get the network interface of data node**

File: libs/ha/ha_common_libs.py [hunk 10]
```diff
@@ -467,7 +469,7 @@ def perform_node_operation(
             resp = sys_obj.node_operation(
                 operation=operation, resource_id=resource_id, force_op=f_opt, storage_off=s_off)
             return resp
-        except Exception as error:
+        except CTException as error:
             LOGGER.error("%s %s: %s",
```
Comment (ankita-93) [inline libs/ha/ha_common_libs.py:473]: same here. Can we use LOGGER.exception ?

File: libs/ha/ha_common_libs_k8s.py (hunk 8 above)
Comment (rahul-hatwar) [reply libs/ha/ha_common_libs_k8s.py:1265]: hmm, because of merge conflicts resolution, this is introduce. Corrected.

File: libs/ha/ha_common_libs_k8s.py (hunk 9 above)
Comment (ankita-93) [reply libs/ha/ha_common_libs_k8s.py:1160]: Is this updated ?

File: libs/ha/ha_common_libs_k8s.py [hunk 11]
```diff
@@ -1456,8 +1461,8 @@ def get_node_resource_ids(node_obj, r_type, n_type=None, node_id=None):
                                          command_suffix=cmd, decode=True)
             resp = literal_eval(resp)
         except IOError as error:
-            LOGGER.error("Failed to get resource IDs for %s", r_type)
-            LOGGER.error("Error in %s: %s", HAK8s.get_node_resource_ids.__name__, error)
+            LOGGER.exception("Failed to get resource IDs for %s", r_type)
```
Comment (ankita-93) [inline libs/ha/ha_common_libs_k8s.py:1464]: this is error only.

File: libs/ha/ha_common_libs_k8s.py (hunk 9 above)
Comment (rahul-hatwar) [reply libs/ha/ha_common_libs_k8s.py:1160]: Not changing this message, as this return is id condition not meet.

**Review by ankita-93 — APPROVED** [review]

File: libs/ha/ha_common_libs_k8s.py [hunk 12]
```diff
@@ -778,18 +779,19 @@ def start_random_mpu(self, event, s3_data, bucket_name, object_name, file_size,
...
                 LOGGER.info("Uploaded part %s", i)
             except CTException as error:
-                LOGGER.error("Error: %s", error)
+                LOGGER.exception("Error: %s", error)
                 if event.is_set():
                     exp_failed_parts.append(i)
                 else:
                     failed_parts.append(i)
-                LOGGER.info("Failed to upload part %s", i)
+                LOGGER.exception("Failed to upload part %s", i)
```
Comment (ApurwaMohite) [inline libs/ha/ha_common_libs_k8s.py:787]: No need to print traceback again. This can be LOGGER.info only as traceback will be printed at line 782.

File: libs/ha/ha_common_libs_k8s.py [hunk 13]
```diff
@@ -711,7 +712,7 @@ def create_bucket_copy_obj(event, s3_test_obj=None, bucket_name=None, object_nam
                         event_clear_flg = False
                         continue
                     failed_bkts.append(bkt_name)
-                LOGGER.info("Failed to copy object to bucket %s", bkt_name)
+                LOGGER.exception("Failed to copy object to bucket %s", bkt_name)
```
Comment (ApurwaMohite) [inline libs/ha/ha_common_libs_k8s.py:715]: No need to print traceback again. This can be LOGGER.info only as traceback will be printed at line 705.

File: tests/ha/test_cluster_shutdown_start.py [hunk 14]
```diff
@@ -84,7 +84,7 @@ def setup_class(cls):
         cls.s3acc_name = cls.s3acc_email = cls.bucket_name = cls.object_name = None
         cls.multipart_obj_path = None
         cls.mgnt_ops = ManagementOPs()
-        cls.system_random = random.SystemRandom()
+        cls.system_random = secrets.SystemRandom()
```
Comment (ApurwaMohite) [inline tests/ha/test_cluster_shutdown_start.py:87]: This should be random.SystemRandom only right? As we want to generate randomness.

**Review by ApurwaMohite — APPROVED** [review]
